Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -68,14 +68,133 @@ public IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
                     {
-                        storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                        switch (eventTypeID)
+                        {
+                            case ExecutionEventId.FileArtifactContentDecided:
+                                storedEvents.Add(FileArtifactContentDecidedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.WorkerList:
+                                storedEvents.Add(WorkerListEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionPerformance:
+                                storedEvents.Add(PipExecutionPerformanceEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DirectoryMembershipHashed:
+                                storedEvents.Add(DirectoryMembershipHashedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessExecutionMonitoringReported:
+                                storedEvents.Add(ProcessExecutionMonitoringReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessFingerprintComputation:
+                                storedEvents.Add(ProcessFingerprintComputationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ExtraEventDataReported:
+                                storedEvents.Add(ExtraEventDataReported.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DependencyViolationReported:
+                                storedEvents.Add(DependencyViolationReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionStepPerformanceReported:
+                                storedEvents.Add(PipExecutionStepPerformanceReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipCacheMiss:
+                                storedEvents.Add(PipCacheMissEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ResourceUsageReported:
+                                storedEvents.Add(StatusReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DominoInvocation:
+                                storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());

```

Comment:
So sorry to ask this, but would you mind using `BuildXLInvocationEvent`

We should use BuildXL everywehre in our naming. bxl is just for our executable name.

Reply:
note: future pr is totally fine.

---
In reply to: [309476346](https://github.com/microsoft/BuildXL/pull/685#discussion_r309476346) [](ancestors = 309476346)

Reply:
So this naming mismatch is due to the BXL (in the corresponding enum) being DominoInvocation. I can change it, but I left it there for similarity purposes

Reply:
Changing it in the next commit.

Reply:
@dannyvv: are you 100% sure it's safe to rename this event?  I vaguely remember a comment somewhere in the code saying that it could not be renamed because of a Kusto table that others depend on (or something along this line).   I'll try to find this comment.

Reply:
This code in [`IExecutionLogTarget`](https://github.com/microsoft/BuildXL/blob/e72fa8841719728183443808432059a92ac0397f/Public/Src/Engine/Scheduler/Tracing/ExecutionLog.Events.cs#L108-L109) is what I meant:
```csharp
        // $Rename: Due to telemetry backend scripts this cannot be renamed to BuildXL
        void DominoInvocation(DominoInvocationEventData data);
```

Reply:
While we cannot rename it in the BXL codebase, I think we can rename it in the Xldb codebase (so users of the lib won't see 'domino'). We might need to ad a comment in 'write db analyzer' explaining why we are using a different name.

---
In reply to: [309813123](https://github.com/microsoft/BuildXL/pull/685#discussion_r309813123) [](ancestors = 309813123)